cranelift/codegen/src/opts/bitops.isle

```diff
@@ -202,3 +202,5 @@
 (rule (simplify (band ty (band ty x y) y)) (band ty x y))
 (rule (simplify (band ty x (band ty x y))) (band ty x y))
 (rule (simplify (band ty y (band ty x y))) (band ty x y))
+
+(rule (simplify (band ty (bxor ty x (bnot ty y)) x)) (band ty x y))
```

Comment:
ISLE doesn't automatically generate rules for matching commutative operations right now, so it may be worth manually enumerating the equivalent commutative versions here, e.g.:

```lisp
(rule (simplify (band ty x (bxor ty x (bnot ty y)))) (band ty x y))
(rule (simplify (band ty (bxor ty (bnot ty y) x) x)) (band ty x y))
(rule (simplify (band ty (bxor ty x (bnot ty y)) x)) (band ty x y))
```